From 451079d61c7c05e103bb3f1a2e7848adc25c4732 Mon Sep 17 00:00:00 2001
From: Martin Jolicoeur <snmpb1@gmail.com>
Date: Mon, 21 Aug 2017 22:17:39 -0400
Subject: [PATCH 08/11] Redefined INVALID_SOCKET to get rid of compiler
 warnings

---
 include/snmp_pp/uxsnmp.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/include/snmp_pp/uxsnmp.h b/include/snmp_pp/uxsnmp.h
index b396260..f4797ee 100644
--- a/include/snmp_pp/uxsnmp.h
+++ b/include/snmp_pp/uxsnmp.h
@@ -44,9 +44,8 @@ namespace Snmp_pp {
 //-----[ internally used defines ]----------------------------------------
 #define MAXNAME 80                   // maximum name length
 #define MAX_ADDR_LEN 10              // maximum address len, ipx is 4+6
-#ifndef INVALID_SOCKET
-#define INVALID_SOCKET ((SnmpSocket)(~0)) // value for invalid socket
-#endif
+#undef INVALID_SOCKET
+#define INVALID_SOCKET (((SnmpSocket)(-1))<0?(SnmpSocket)(-1):((SnmpSocket)(~0))) // value for invalid socket
 
 //-----[ async defines for engine ]---------------------------------------
 #define sNMP_PDU_GET_ASYNC       21
-- 
2.17.1

